lesson18/src/js/main.js

```diff
@@ -0,0 +1,299 @@
+
+  if (currentNum < pagenationTotal && buttonNext) {
+    pagenationCurrent.textContent = currentNum += 1;
+  } else if (currentNum <= pagenationTotal && buttonPrev) {
+    pagenationCurrent.textContent = currentNum -= 1;
+  }
+}
+
+function switchImg(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  const firstSlide = document.querySelector('[data-slide-index="1"]');
+  if(active[target]){
+    active[target].classList.add("is-active");
+  }else{
+    firstSlide.classList.add("is-active");
+  }
+}
+
+function switcPagenationBullet(target){
+  const pagenationBullet = document.querySelector('[aria-current="true"');
+  pagenationBullet.setAttribute('aria-current', false);
+  const firstPagenationBullet = document.querySelector('[data-pagenation-index="1"]');
+
+  if(pagenationBullet[target]){
+    pagenationBullet[target].setAttribute('aria-current',true);
+  }else{
+    firstPagenationBullet.setAttribute('aria-current',true);
+  }
+}
+
+function clickPagenationBullet(){
+  const pagenationBullet = [...document.getElementsByClassName("bar")];
+  pagenationBullet.forEach((targetButton) => {
+    targetButton.addEventListener('click',(e) => {
+      const activePagenationBullet = document.querySelector('[aria-current="true"]');
+      const clickedPagenationBulletNo = e.currentTarget.getAttribute('data-pagenation-index');
+      activePagenationBullet.setAttribute('aria-current','false');
+      e.currentTarget.setAttribute('aria-current','true');
+      document.getElementById('js-current').textContent = clickedPagenationBulletNo;
+
+      const activeSlide = document.getElementsByClassName('slider__item is-active');
+      activeSlide[0].classList.remove('is-active');
+      document.querySelector(`[data-slide-index="${clickedPagenationBulletNo}"]`).classList.add('is-active');
+
+      const pagenationCurrent = document.getElementById("js-current");
+      pagenationCurrent.textContent = clickedPagenationBulletNo;
+      toggleButtonDisabled();
+      resetAutoPlaySlide();
+    })
+  })
+
+}
+
+function toggleButtonDisabled() {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+  const activeImage = document.querySelector(".is-active");
+  const currentIndex = Number(activeImage.dataset.slideIndex);
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+  prevButton.disabled = currentIndex === firstIndex;
+  nextButton.disabled = currentIndex === lastIndex;
+}
+
+function renderSlidContents(slideImageData){
+  renderSlideArea();
+  renderSlideImage(slideImageData);
+  renderPagenation(slideImageData.length);
+  renderPagenationBullet(slideImageData.length);
+  renderSlideButton();
+  addSwitchButtonEvent();
+  clickPagenationBullet();
+  toggleButtonDisabled();
+  autoPlayslide();
+}
```

Comment:
IMO
I think it is a good idea to separate or rename the functions since they are also executing other functions than render.

Reply:
The function name and execution contents do not match!
It is divided into a slide drawing function and a function for adding a slide function.

[faa8f94](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/faa8f94309187bac163331dada67e6ca1ac695d6)

> IMO I think the auto function is not reset when the arrow button is pressed
> 
> https://gyazo.com/72821e9513d4182708b1e260debb9818

Thank you for your review!
Thank you for politely making a video.
Since autoplay was not reset when the button was clicked,
Added a reset function.

[11769c8](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/11769c8c6a74de3bdc1b178592a8a114e527250d)